File: _CodeTopics/LeetCode_contest/weekly/weekly2021/263-[TI10最后一天的比赛]/263_1.py
class Solution(object):
    def areNumbersAscending(self, s):
        """
        :type s: str
        :rtype: bool
        """
        
        l = s.split()
        tmp = []
        for token in l:
            if token[0].isdigit():
                tmp.append(int(token))
        return tmp == sorted(list(set(tmp)))
    
"""
https://leetcode-cn.com/submissions/detail/229582321/

97 / 97 个通过测试用例
状态：通过
执行用时: 20 ms
内存消耗: 13.3 MB
"""

File: _CodeTopics/LeetCode/2001-2200/002042--weekly263_1/tran/263_1.cpp
class Solution {
public:
    bool areNumbersAscending(string s) {

        // 第 263 场周赛第一题
        // https://leetcode.cn/submissions/detail/229582321/

        vector<string> l = {};
        stringstream ss(s);
        string str;
        while (ss >> str) {
            l.emplace_back(str);
        }

        vector<string> tmp = {};
        int currNum = 0;
        for (auto token : l) {
            if (isdigit(token[0])) {
                if (stoi(token) > currNum) {
                    currNum = stoi(token);
                } else {
                    return false;
                }
            }
        }
        return true;
    }
};

/*
https://leetcode.cn/submissions/detail/392541479/

执行用时：
4 ms
, 在所有 C++ 提交中击败了
30.38%
的用户
内存消耗：
6.5 MB
, 在所有 C++ 提交中击败了
5.06%
的用户
通过测试用例：
98 / 98
*/
